src/components/Products/ProductsList.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { useState, useEffect } from "react";
+import "./ProductsList.css";
+import ProductThumb  from "./ProductThumb";
+import Sidebar from "../Sidebar/Sidebar";
+import Products from "../../mocks/en-us/products.json";
+
+function ProductsList() {
+  let [selectedCategory, setSelectedCategory] = useState("")
+  let [data] = useState(Products.results);
+  let [products, setProducts] = useState([]);
+  useEffect(() => {
+    setProducts(selectedCategory.length > 0 ? data.filter(d => d.data.category.id === selectedCategory) : data);
+  },[data, selectedCategory]);
```

Comment:
Creo que esta sección podría retrabajarse para una mejor lectura. Hay varias cosas que se les puede hacer fix.

1.  Las variables puedes declararlas como `const` porque al usar hooks siempre obtenemos una instancia nueva, no estamos haciendo ninguna asignación con el operador `=`.
2. Si defines tu state de `selectedCategory` como null en el hook la condición del ternario va ser más fácil de definir.

Quedaría algo así:
```suggestion
  const [selectedCategory, setSelectedCategory] = useState(null)
  const [data] = useState(Products.results);
  const [products, setProducts] = useState([]);
  useEffect(() => {
    setProducts( selectedCategory && data.filter(d => d.data.category.id === selectedCategory) : data);
  },[data, selectedCategory]);
```